labs/data_encoding_health.py

```diff
@@ -0,0 +1,116 @@
+
+import pandas as pd
+from helpers.dslabs_functions import get_variable_types, encode_cyclic_variables, dummify
+import numpy as np
+
+
+def calulate_mean_age(age_categories):
+    mean_ages = []
+    for value in age_categories:
+        split = value.split()
+        age1, age2 = int(split[1]), split[3]
+        if age2.isdigit():
+            mean_ages.append((age1+int(age2))/2)
+        else:
+            mean_ages.append(85)
+    return mean_ages
+
+def yes_no_mapping(answer):
+    if pd.isna(answer):
+        return np.nan
+    elif answer.startswith('Yes'):
+        return 1
+    elif answer.startswith('No'):
+        return 0
+    else:
+        return np.nan  # or a specific code for answers that do not start with Yes/No
+
+
+def parse_location_file(file_path):
+    location_mapping = {}
+
+    with open(file_path, 'r') as file:
+        next(file)  # Skip the header line
+        for line in file:
+            # Remove quotation marks and split the line
+            parts = line.strip().replace('"', '').split(':')
+            state = parts[0].strip()
+            coords = parts[1].strip().strip('[]').split(',')
+            longitude = float(coords[0].strip())
+            latitude = float(coords[1].strip())
+
+            location_mapping[state] = {'Latitude': latitude, 'Longitude': longitude}
+
+    return location_mapping
+
+def add_coordinates(df, mapping):
+    # Extract latitude and longitude from the mapping
+    df['Latitude'] = df['State'].map(lambda x: mapping[x]['Latitude'] if x in mapping else None)
+    df['Longitude'] = df['State'].map(lambda x: mapping[x]['Longitude'] if x in mapping else None)
+    return df
+
+def encode_symbolic(df, encoding):
+    
+    df = pd.get_dummies(df, columns=['RaceEthnicityCategory'], dummy_na=False)
```

Comment:
Isabelle had an idea here:
Instead of dummifying the data, we can look at which ethnicity has the most people and then encode that as 0, the second most occuring ethnicity is 1, then third as 2. No idea how to handle the in between values, as black (no hispanic) and hispanic would then be 0.5 in between.

Could be interesting to look at, to avoid dummifying the data

Reply:
Sceptic to this, and don't know exactly how it works. Might ask the teacher today